content/news/2024-02-03-bevy-0.13/index.md

```diff
@@ -529,11 +529,15 @@ world.register_system(|world: &mut World| { /* do anything */ });
 
 All these improvements round out one-shot systems significantly and they should now behave normally in any Bevy context.
 
-## WGPU Upgrade
+## WGPU Upgrade and Renderer Improvements
 
-<div class="release-feature-authors">authors: @TODO</div>
+<div class="release-feature-authors">authors: @Elabajaba</div>
 
-TODO.
+We upgrade from wgpu 0.17 in Bevy 0.12 to wgpu 0.19, which includes the long awaited wgpu [arcanization](https://gfx-rs.github.io/2023/11/24/arcanization.html) that allows us to do things like compile shaders in parallel in the background to avoid shader compilation stutters, and to multithread draw call creation for better performance.
+
+Due to changes in wgpu 0.19, we've added a new `webgpu` feature to Bevy that is now required when doing WebAssembly builds targeting WebGPU. Disabling the `webgl2` feature is no longer required when targeting WebGPU, but the new `webgpu` feature currently overrides the `webgl2` feature when enabled. Library authors, please do not enable the `webgpu` feature by default. In the future we plan on allowing you to target both WebGL2 and WebGPU in the same WebAssembly binary, but it requires reworking parts of the renderer where we're relying on compile time constants when targeting `webgl2`, and adding a way to choose the renderer's backend at runtime on web.
+
+As usual, there's been some changes that may cause issues for custom shaders. We've swapped the material and mesh bind groups, so that mesh data is now in bind group 1, and material data is in bind group 2. This greatly improved our draw call batching when combined with changing our sorting order from using the distance from the camera, to sorting by the pipeline and mesh, which can lead to much better CPU performance for rendering. We've also removed the `get_instance_index` function in shaders, as it was only required to workaround an upstream bug that has been fixed in wgpu 0.19. For other shader or rendering changes, please see the [migration guide](/learn/migration-guides/0.12-0.13/) and [wgpu's changelog](https://github.com/gfx-rs/wgpu/blob/v0.19/CHANGELOG.md).
```

Comment:
```suggestion
As usual, there's been some changes that may cause issues for custom shaders. We've swapped the material and mesh bind groups, so that mesh data is now in bind group 1, and material data is in bind group 2. This greatly improved our draw call batching when combined with changing our sorting order from using the distance from the camera, to sorting by the pipeline and mesh, which can lead to greater CPU performance for rendering. We've also removed the `get_instance_index` function in shaders, as it was only required to workaround an upstream bug that has been fixed in wgpu 0.19. For other shader or rendering changes, please see the [migration guide](/learn/migration-guides/0.12-0.13/) and [wgpu's changelog](https://github.com/gfx-rs/wgpu/blob/v0.19/CHANGELOG.md).
```